### PR TITLE
Update to PureScript v0.15.0

### PR DESCRIPTION
**Pull request by sigma-andex**
**Description of the change**

Backlinking to purescript/purescript#4244

Migrates FFI to ES modules

---

**Checklist:**

- [x] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0000)")
- [x] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by sigma-andex** [comment]
🏓 @JordanMartinez

File: test/Main.purs [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import Prelude
+
+import Effect (Effect)
+import Node.URL (format, parse, parseQueryString, toQueryString)
+import Test.Assert (assertEqual)
+
+main ∷ Effect Unit
+main = do
+  assertEqual { expected: "http://example.com/", actual: format $ parse "http://example.com/" }
+  assertEqual { expected: "foo=42", actual: toQueryString $ parseQueryString "foo=42" }
```
Comment (thomashoneyman) [inline test/Main.purs]:
```suggestion
  assertEqual { expected: "foo=42", actual: toQueryString $ parseQueryString "foo=42" }

```

**Review by thomashoneyman — APPROVED** [review]

**Comment by thomashoneyman** [comment]
@sigma-andex Now that there are tests, do you mind adding a `test` script to run them in the package.json?

**Comment by thomashoneyman** [comment]
Also, looks like we've got a linting error.

**Comment by thomashoneyman** [comment]
Whoops, one more thing — can you add this to the CHANGELOG?

**Review by JordanMartinez — CHANGES_REQUESTED** [review]
Yeah, just need the changelog.

**Review by thomashoneyman — APPROVED** [review]

File: bower.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
     "url": "https://github.com/purescript-node/purescript-node-url.git"
   },
   "dependencies": {
-    "purescript-nullable": "^5.0.0"
+    "purescript-nullable": "main",
+    "purescript-assert": "master"
```
Comment (JordanMartinez) [inline bower.json:18]: Shouldn't this be a devDependency?

**Review by JordanMartinez — CHANGES_REQUESTED** [review]
I think 'assert' should be a dev dependency in bower.json

**Pull request merged**